src/HttpClient.ts

```diff
@@ -197,12 +197,9 @@ export class HttpClient extends EventEmitter {
       }
       requestUrl = new URL(url);
     } else {
-      if (!url.searchParams) {
-        // url maybe url.parse(url) object in urllib2
-        requestUrl = new URL(urlFormat(url));
-      } else {
-        requestUrl = url;
-      }
+      // url maybe url.parse(url) object in urllib2
+      // or even if not, we clone to avoid mutating it
+      requestUrl = new URL(urlFormat(url));
```

Comment:
https://nodejs.org/dist/latest-v18.x/docs/api/url.html#urlformaturlobject
urlFormat is only for the `url.parse()` return object. `
If the url is a URL instance, I think we should use `new URL(url.toString())` is better.

And the `urlformat()` is `Legacy`, it should not be used for non-compatibility issues.

Reply:
I see